pkg/coscheduling/apis/podgroup/v1alpha1/types.go

```diff
@@ -0,0 +1,134 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PodGroupPhase is the phase of a pod group at the current time.
+type PodGroupPhase string
+
+// These are the valid phase of podGroups.
+const (
+	// PodPending means the pod group has been accepted by the system, but scheduler can not allocate
+	// enough resources to it.
+	PodGroupPending PodGroupPhase = "Pending"
+
+	// PodRunning means `spec.minMember` pods of PodGroups has been in running phase.
+	PodGroupRunning PodGroupPhase = "Running"
+
+	// PreScheduling means all of pods has been are waiting to be scheduled, enqueue waitingPod
+	PodGroupPreScheduling PodGroupPhase = "PreScheduling"
+
+	// PodRunning means some of pods has been scheduling in running phase but have not reach the `spec.
+	// minMember` pods of PodGroups.
+	PodGroupScheduling PodGroupPhase = "Scheduling"
+
+	// PodScheduled means `spec.minMember` pods of PodGroups have been scheduled finished and pods have been in running
+	// phase.
+	PodGroupScheduled PodGroupPhase = "Scheduled"
+
+	// PodGroupUnknown means part of `spec.minMember` pods are running but the other part can not
+	// be scheduled, e.g. not enough resource; scheduler will wait for related controller to recover it.
+	PodGroupUnknown PodGroupPhase = "Unknown"
+
+	// PodGroupFinish means all of `spec.minMember` pods are successfully.
+	PodGroupFinished PodGroupPhase = "Finished"
+
+	// PodGroupFailed means at least one of `spec.minMember` pods is failed.
+	PodGroupFailed PodGroupPhase = "Failed"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodGroup is a collection of Pod; used for batch workload.
+type PodGroup struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the pod group.
+	// +optional
+	Spec PodGroupSpec `json:"spec,omitempty"`
+
+	// Status represents the current information about a pod group.
+	// This data may not be up to date.
+	// +optional
+	Status PodGroupStatus `json:"status,omitempty"`
+}
+
+// PodGroupSpec represents the template of a pod group.
+type PodGroupSpec struct {
+	// MinMember defines the minimal number of members/tasks to run the pod group;
+	// if there's not enough resources to start all tasks, the scheduler
+	// will not start anyone.
+	MinMember uint32 `json:"minMember,omitempty"`
+
+	// MinResources defines the minimal resource of members/tasks to run the pod group;
+	// if there's not enough resources to start all tasks, the scheduler
+	// will not start anyone.
+	MinResources *v1.ResourceList `json:"minResources,omitempty"`
```

Comment:
As comments before, MinResource was going to removed.
https://github.com/kubernetes-sigs/scheduler-plugins/pull/42#discussion_r478137253

Reply:
https://github.com/kubernetes-sigs/scheduler-plugins/pull/42#discussion_r481790970, this it  the latest.

Reply:
> I'm confused about how to set the MinResource. It's the total minimal resource of the PodGroup or the pod?
> yes
---
Yes？  the PodGroup or the pod？

Reply:
> > I'm confused about how to set the MinResource. It's the total minimal resource of the PodGroup or the pod?
> > yes
> 
> Yes？ the PodGroup or the pod？

We have discussed about this question offline. 

minimal resource of `PodGroup`

Reply:
Ok. It's better to change like this:

// MinResources defines the minimal resource of members/tasks to run the pod group; ---> // MinResources defines the minimal resource to run the pod group;